nntrainer/graph/network_graph.cpp

```diff
@@ -393,56 +393,122 @@ sharedConstTensors NetworkGraph::incremental_forwarding(
   return out;
 }
 
-void NetworkGraph::backwarding(
+bool NetworkGraph::backwarding(
```

Comment:
This is the index of whether the gradient has Nan or Inf. If it has, we do need to compute the loss scale factor again and compute the derivatives again so that, the gradient is valid. I will add the comment in header. You can also check the backwarding in neuralnet.cpp.